hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/hbarapprove/HbarApproveTranslator.java

```diff
@@ -104,7 +104,7 @@ private CryptoApproveAllowanceTransactionBody cryptoApproveTransactionBody(
                 .cryptoAllowances(CryptoAllowance.newBuilder()
                         .owner(owner)
                         .spender(operatorId)
-                        .amount(amount.longValue())
+                        .amount(amount.longValueExact())
```

Comment:
ha! I guess that means these tests were a good idea, TY. (And also for the next few files)

Reply:
agreed